securesystemslib/storage.py

```diff
@@ -219,12 +223,28 @@ def __exit__(self, exc_type, exc_val, traceback):
   get = GetFile
 
 
-  def put(self, fileobj, filepath):
+  def put(self, fileobj, filepath, permissions=None):
     # If we are passed an open file, seek to the beginning such that we are
     # copying the entire contents
     if not fileobj.closed:
       fileobj.seek(0)
 
+    # When calculating file permissions, the current OS umask value is first
+    # masked out. If custom permissions are set, we calculate a new umask
+    # followed by a call to open() with default permissions (0o777).
+    # This trick is done since open() does not support mode(permissions) as
+    # input parameter, alternatively os.open() can be used.
```

Comment:
I had a look again. If we use `os.open()` its documentation says _"When computing mode, the current umask value is first masked out."_ Meaning if we want to avoid race conditions caused by `os.umask`, we can pass the desired permissions to `os.open` but the resulting file permissions maybe restricted even further due to the applied mask.
 
This is not necessarily a bad thing for this concrete case where we want to restrict keys permissions. We can put the same line as a disclaimer in the docs.

Reply:
Yep :facepalm: